tests/core/full_node/test_full_node.py

```diff
@@ -165,7 +168,16 @@ async def check_transaction_confirmed(transaction) -> bool:
         # Confirm generator is not compressed
         program: Optional[SerializedProgram] = (await full_node_1.get_all_full_blocks())[-1].transactions_generator
         assert program is not None
-        assert detect_potential_template_generator(uint32(5), program) is not None
+        if consensus_mode == ConsensusMode.HARD_FORK_2_0:
+            # after the hard fork we don't use this compression mechanism
+            # anymore, we use CLVM backrefs in the encoding instead
+            check = lambda x: x is None  # noqa: 731
+            check_num_blocks = lambda x: x == 0  # noqa: 731
+        else:
+            check = lambda x: x is not None  # noqa: 731
+            check_num_blocks = lambda x: x > 0  # noqa: 731
```

Comment:
Not so excited about this lambda blocks and the "noqa"s  but i guess its fine :) Maybe rename `check` to `check_none`.

Reply:
flake8 complains about assigning lambdas to variables and wants you to use `def` instead. mypy doesn't like that you have conditional and different `def`s (and defs take a lot more space)